### DISCUSSION TITLE
Difference between network requests (Docker issues)

### DISCUSSION THREAD

**Discussion by Richard-Woessner** [discordjs/discord.js#10131] [Q&A] [answered]
I've been attempting to make an extremely simple discord bot. The only tasks my bot would do are `Message.React()` and `Message.Respond()` and at an objective level my bot is finished. It does everything I want the issues to come when I try to put it into a docker container.

My docker container is using the host network so it shouldn't deal with porting or anything.

When I put it into a docker container the bot successfully logs in and will trigger when a message is sent to the chat, but the issue arrives when the bot attempts to communicate back.

`Message.react ` and `Message.reply` causes a TIMEOUT error and I'm trying to find out why. I'm newer to docker so I can only assume it's an authentication issue.

My next attempt is to try to see what the difference between the client "connecting" and the client "responding".

If anyone has any ideas, please let me know, before I start losing sleep.

**Comment by Qjuh** [comment]
The difference is that "connecting" uses the gateway which is a websocket connection. While "responding" is a call to the REST API with a regular HTTP request.

**Reply by Richard-Woessner** [reply]
Thank you any chance you could send me an example of what Discords API requests look like?

**Reply by Qjuh** [reply]
https://discord.com/developers/docs

**Reply by Richard-Woessner** [reply, marked as answer]
Thank you, I found the "issue". Basically, it turns out that docker doesn't run the greatest on Mac. I assumed that it ran fine because the Dockerfile built rather quickly. After a lot of googling, I read about this issue with Docker on Mac, and just tested it on my Linux server and it runs fine. 

So I just started developing locally and just used docker to put it in a container on my server and it works fine.

Thanks
